client/src/actions/auth.js

```diff
@@ -2,68 +2,94 @@
  * This file houses our auth-related Action Creators.
  */
 import * as api from '../api';
-import { AUTH_SIGN_IN_REQUEST, AUTH_SIGN_IN_SUCCESS, AUTH_SIGN_OUT, AUTH_SIGN_IN_FAIL, FETCH_TEAM_INFO, CLEAR_ERROR } from './types';
+import {
+  AUTH_SIGN_IN_REQUEST,
+  AUTH_SIGN_IN_SUCCESS,
+  AUTH_SIGN_OUT,
+  AUTH_SIGN_IN_FAIL,
+  FETCH_TEAM_INFO,
+  CLEAR_ERROR,
+  FETCH_WEBSITE_INFO,
+} from './types';
 import { errorActionGlobalCreator } from '../error/errorReduxFunctions';
 
-
 /**
  * This action creator will be called when a user signs in.
  *
  * @param authData data associated to the authentication response.
  * @returns an action of type AUTH_SIGN_IN with the payload as the authData.
  */
-export const signIn = (authData) => async(dispatch) => {
-  try{
-    dispatch ({
-      type: AUTH_SIGN_IN_REQUEST
-    })
-    const { data } = await api.loginTeam(authData)
+export const signIn = (authData) => async (dispatch) => {
+  let teamId;
+  try {
     dispatch({
-      type: AUTH_SIGN_IN_SUCCESS
-    })
+      type: AUTH_SIGN_IN_REQUEST,
+    });
+    const { data } = await api.loginTeam(authData);
+    teamId = data._id;
+    dispatch({
+      type: AUTH_SIGN_IN_SUCCESS,
+    });
     dispatch({
       type: FETCH_TEAM_INFO,
-      payload: data
-    })
-  } catch (error){
+      payload: data,
+    });
+  } catch (error) {
     dispatch({
-      type: AUTH_SIGN_IN_FAIL
-    })
+      type: AUTH_SIGN_IN_FAIL,
+    });
     dispatch(errorActionGlobalCreator(error));
   }
+
+  await api
```

Comment:
I don't it is a proper way to put the getWebsiteInfo api request here as it will only be called when the user first log in, and once the user refreshes the page after login, all the info stored in the redux state would be gone and it is no way to retrieve this info again unless the user log in again.

Reply:
True... where would you think this should be placed to avoid data being deleted when the page is reloaded. I'm thinking we can have it in the dashboard, but wanted to hear your thoughts @pmyip3

Reply:
Yes, we could put the API request inside an useEffect hook in the Dashboard component, so every time the dashboard page gets refreshed, the request will be sent to the backend again. But even if we are not on the Dashboard page and we try to refresh the page, we still want to send that API request as the relevant pages should be on the sidebar. 

So I think we can just also return the website Info in the authoorizeJWT function as this function will get called every time the private route gets refreshed. 
![image](https://user-images.githubusercontent.com/17608976/127650131-184f7f54-9c32-474c-a7fd-4d89c8a1b556.png)


In this way, the getTeamJWT API request would return both team_info and website_info, then we can just dispatch two actions with the corresponding results. 
![image](https://user-images.githubusercontent.com/17608976/127650310-ba40130a-20e6-461a-917a-c432d7aafdc9.png)

Reply:
The corresponding backend endpoints for getTeamJWT() need to query both the teams and website collection as well in order to return both information. 

Just my thought, might require a lot of changes though. Feel free to comment on it.